res/controllers/Traktor-Kontrol-S2-MK3-hid-scripts.js

```diff
@@ -0,0 +1,974 @@
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "eject", field.value);
+        // Reset eject button
+        engine.setValue(field.group, "eject", 0);
+    } else {
+        engine.setValue(field.group, "LoadSelectedTrack", field.value);
+    }
+};
+
+TraktorS2MK3.addTrackHandler = function (field) {
+    TraktorS2MK3.outputHandler(field.value, field.group, "addTrack");
+
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue("[Library]", "AutoDjAddTop", field.value);
+    } else {
+        engine.setValue("[Library]", "AutoDjAddBottom", field.value);
+    }
+};
+
+TraktorS2MK3.maximizeLibraryHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    var maximize = engine.getValue("[Master]", "maximize_library");
+    engine.setValue("[Master]", "maximize_library", !maximize);
+};
+
+TraktorS2MK3.selectLoopHandler = function (field) {
+    if ((field.value + 1) % 16 === TraktorS2MK3.loopSizeState[field.group]) {
+        engine.setValue(field.group, "loop_halve", 1);
+    } else {
+        engine.setValue(field.group, "loop_double", 1);
+    }
+    TraktorS2MK3.loopSizeState[field.group] = field.value;
+};
+
+TraktorS2MK3.activateLoopHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        engine.setValue(field.group, "reloop_andstop", field.value);
+    } else {
+        engine.setValue(field.group, "reloop_toggle", field.value);
+    }
+};
+
+TraktorS2MK3.beatjumpHandler = function (field) {
+    var delta = 1;
+    if ((field.value + 1) % 16 === TraktorS2MK3.beatjumpState[field.group]) {
+        delta = -1;
+    }
+
+    if (TraktorS2MK3.shiftPressed[field.group]) {
+        var beatjump_size = engine.getValue(field.group, "beatjump_size");
+        if (delta > 0) {
+            engine.setValue(field.group, "beatjump_size", beatjump_size * 2);
+        } else {
+            engine.setValue(field.group, "beatjump_size", beatjump_size / 2);
+        }
+    } else {
+        if (delta < 0) {
+            engine.setValue(field.group, "beatjump_backward", 1);
+            engine.setValue(field.group, "beatjump_backward", 0);
+        } else {
+            engine.setValue(field.group, "beatjump_forward", 1);
+            engine.setValue(field.group, "beatjump_forward", 0);
+        }
+    }
+
+    TraktorS2MK3.beatjumpState[field.group] = field.value;
+};
+
+TraktorS2MK3.quantizeHandler = function (field) {
+    if (field.value === 0) {
+        return;
+    }
+
+    var res = !(engine.getValue("[Channel1]", "quantize") && engine.getValue("[Channel2]", "quantize"));
+    engine.setValue("[Channel1]", "quantize", res);
+    engine.setValue("[Channel2]", "quantize", res);
+    TraktorS2MK3.outputHandler(res, field.group, "quantize");
+};
```

Comment:
You could make this shift dependent: w/o shift: left deck, w/ shift: right deck. I'm not sure if that would improve things though, so feel free to ignore this ;-)

Reply:
I think this would confuse people, since there is only one quantize button on the controller and its not obvious which deck is meant.

Reply:
OK.